### ISSUE TITLE
New iOS architecture: arm64e

### ISSUE THREAD

**Issue by theodelrieu** [conan-io/conan#4134]
Hello,

Apple recently added a new iOS architecture: arm64e.

From this [SO thread](https://stackoverflow.com/questions/52624308/xcode-arm64-vs-arm64e/52624441), it seems that it uses ARM v8.3. It might be a good idea to introduce it at the same time than the one discussed in #3961.

I do not know if adding `armv8.3` is the way to go, I'm far from being an expert in CPU architectures.

**Comment by lasote** [comment]
I'm investigating a bit. Thanks for telling @theodelrieu !

**Comment by lasote** [comment]
Yes, I agree. Looks like `ARMv8.3` is aka `arm64e`. As we already have a setting `armv8` instead of the corresponding `arm64` we should add `armv8.3`.

**Comment by lasote** [comment]
So the feature will consist of:

- Introduce the new setting `armv8.3` in the default settings.yml (arch, arch_build)
- Review all build helpers/flags where we do something with the arm architecture (search for it): E.g: cmake_flags.py for example, b2.py, win.py: vcvars support?, apply.py, oss.py

**Comment by theodelrieu** [comment]
Yes, seems correct, we also have to update the Apple tools to detect the new arch
